File: Ordenamiento/InsertionSort.java
public class InsertionSort{

	int[] arr = {50,17,48,23,8,60};
	static int i;
	
	public static void main(String[] args){
		InsertionSort is = new InsertionSort();
		is.insertionSort(is.arr); //Ordenamiento InsertionSort
		
		for (i=0; i<is.arr.length; i++){
			System.out.println(is.arr[i]);
		}
	}

	public void insertionSort(int[] A){
		int i, k, x, nelem = A.length;

		// Desde el segundo elemento
		for (i=1; i<nelem; i++){
			x=A[i];
			k=i-1;

			// Para k=-1 se ha alcanzado el extremo izquierdo
			while(k>=0 && x<A[k]){
				A[k+1]=A[k]; //Hace hueco para insertar
				k--;
			}
			A[k+1]=x; //Inserta x en su lugar
		}
	}

}

File: EDAII x Python/Ordenamiento/InsertionSort.py
class InsertionSort:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase.'
		self.arr = arr

	def insertionSort(self):
		'Algoritmo de ordenamiento.'
		nelem = len(self.arr)

		for i in range(1,nelem):
			x = self.arr[i]
			k = i-1

			# Para k=-1 se ha alcanzado el extremo izquierdo
			while k>=0 and x<self.arr[k]:
				self.arr[k+1] = self.arr[k]	# Hace hueco para insertar
				k-=1

			self.arr[k+1] = x

# Ejecutando el objeto
programa = InsertionSort([50,17,48,23,8,60])
programa.insertionSort()
print(programa.arr)